### PR TITLE
Adding view hint paths config option.

### PR DESCRIPTION
**Pull request by incraigulous**
Reference: https://github.com/tightenco/jigsaw/issues/370

### CODE REVIEW

**Comment by incraigulous** [comment]
I'm not sure where to add my documentation, but this adds a config option to add blade hint paths (namespaces): 


```
<?php

return [
    'production' => false,
    'baseUrl' => '',
    'collections' => [],
    'viewHintPaths' => [
        'email:templates' => __DIR__.'/source/_layouts',
        'email:components' => __DIR__.'/source/_components',
        'email:partials' => __DIR__.'/source/_partials'
    ]
];
```

File: tests/ViewRendererTest.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        new ViewRenderer($mock, \Mockery::mock(BladeCompiler::class), [
+            'viewHintPaths' => [
+                'view::hint' => 'path'
+            ]
+        ]);
+        $this->addToAssertionCount(
+            \Mockery::getContainer()->mockery_getExpectationCount()
+        );
+        \Mockery::close();
+    }
```
Comment (damiani) [inline tests/ViewRendererTest.php]: Throughout this test, can you update to follow style conventions from other tests?

- annotate with `@test`, and camel case test names:

```
    /**
     * @test
     */
    public function it_registers_hint_paths()
    {
```

- import `Mockery` with a `use` statement, and remove the trailing `\`
- no need to call `Mockery::close()`, that's handled in `tearDown()`

File: tests/ViewRendererTest.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+
+namespace Tests;
+
+
```
Comment (damiani) [inline tests/ViewRendererTest.php]:
```suggestion
```

File: tests/ViewRendererTest.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+
```
Comment (damiani) [inline tests/ViewRendererTest.php]:
```suggestion
```

File: tests/ViewRendererTest.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+namespace Tests;
+
+
+use Illuminate\View\Compilers\BladeCompiler;
+use Illuminate\View\Factory;
+use TightenCo\Jigsaw\View\ViewRenderer;
+
+class ViewRendererTest extends TestCase
+{
+    public function testItRegisteresHintPaths()
```
Comment (damiani) [inline tests/ViewRendererTest.php]:
```suggestion
    public function testItRegistersHintPaths()
```

File: tests/ViewRendererTest.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        $mock->shouldReceive('addExtension');
+        new ViewRenderer($mock, \Mockery::mock(BladeCompiler::class), [
+            'viewHintPaths' => []
+        ]);
+        $this->addToAssertionCount(
+            \Mockery::getContainer()->mockery_getExpectationCount()
+        );
+        \Mockery::close();
+    }
+}
```
Comment (damiani) [inline tests/ViewRendererTest.php]:
```suggestion
}

```

**Comment by incraigulous** [comment]
@damiani Thanks for the review! The pull request is updated with your suggestions. I amended the initial commit to keep your timeline clean. Let me know if you would like any further changes!

**Comment by damiani** [comment]
@incraigulous Excellent, thanks for the contribution!

**Pull request merged**